lib/refile/s3.rb

```diff
@@ -45,7 +45,6 @@ def initialize(region:, bucket:, max_size: nil, prefix: nil, hasher: Refile::Ran
       @s3 = Aws::S3::Resource.new @s3_options
       credentials = @s3.client.config.credentials
       raise S3CredentialsError unless credentials
-      @access_key_id = credentials.access_key_id
```

Comment:
could just change to `@access_key_id = credentials[:access_key_id]`? See [#credentials documentation](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/Core/Configuration.html#credentials-instance_method), but this might not work for all Credential Providers..